gotfparse/pkg/converter/converter.go

```diff
@@ -36,118 +30,214 @@ type terraformMeta struct {
 func (t *terraformConverter) VisitJSON() *gabs.Container {
 	jsonOut := gabs.New()
 
-	var visitor func(b *terraform.Block, parentKey string)
+	for _, m := range t.modules {
+		t.visitModule(m, jsonOut)
+	}
 
-	visitor = func(b *terraform.Block, parentKey string) {
-		blockName := b.GetMetadata().String()
+	return jsonOut
+}
 
-		blockJSON := gabs.New()
-		arrayKey := blockName
+func (t *terraformConverter) shouldWalk(b *terraform.Block) bool {
+	switch b.Type() {
+	case "locals", "output", "terraform", "variable":
+		return false
+	}
 
-		if parentKey != "" {
-			if parentKey != blockName {
-				arrayKey = fmt.Sprintf("%s.%s", parentKey, blockName)
-			} else {
-				arrayKey = parentKey
-			}
+	return true
+}
+
+func (t *terraformConverter) isResource(b *terraform.Block) bool {
+	return b.Type() == "resource"
+}
+
+func (t *terraformConverter) buildBlock(b *terraform.Block) map[string]interface{} {
+	obj := make(map[string]interface{})
+
+	process := handleDynamicBlocks(func(block *terraform.Block) {
+		key := block.Type()
+		value, ok := obj[key]
+		if !ok {
+			value = make([]interface{}, 0)
+			obj[key] = value
 		}
+		list := value.([]interface{})
+		obj[key] = append(list, t.buildBlock(block))
+	})
```

Comment:
I don't mind the closure, but I think in-lining this is stretching the limits of what is a readable function argument and this should be extracted.